lit.ts

```diff
@@ -99,6 +99,52 @@ export async function storeConditionWithSigner(
 	return tx;
 }
 
+export async function mintPKPV2({
+	keyType,
+	permittedAuthMethodTypes,
+	permittedAuthMethodIds,
+	permittedAuthMethodPubkeys,
+	permittedAuthMethodScopes,
+	addPkpEthAddressAsPermittedAddress,
+	sendPkpToItself,
+}: {
+	keyType: string;
+	permittedAuthMethodTypes: string[];
+	permittedAuthMethodIds: string[];
+	permittedAuthMethodPubkeys: string[];
+	permittedAuthMethodScopes: string[][];
+	addPkpEthAddressAsPermittedAddress: boolean;
+	sendPkpToItself: boolean;
+}): Promise<ethers.Transaction> {
+	console.log(
+		"In mintPKPV2",
+		keyType,
+		permittedAuthMethodTypes,
+		permittedAuthMethodIds,
+		permittedAuthMethodPubkeys,
+		permittedAuthMethodScopes,
+		addPkpEthAddressAsPermittedAddress,
+		sendPkpToItself,
+	);
+	const pkpHelper = getPkpHelperContract();
+	const pkpNft = getPkpNftContract();
+
+	// first get mint cost
+	const mintCost = await pkpNft.mintCost();
+	const tx = await pkpHelper.mintNextAndAddAuthMethods(
+		keyType,
+		permittedAuthMethodTypes,
+		permittedAuthMethodIds,
+		permittedAuthMethodPubkeys,
+		permittedAuthMethodScopes,
+		addPkpEthAddressAsPermittedAddress,
+		sendPkpToItself,
+		{ value: mintCost },
+	);
+	console.log("tx", tx);
+	return tx;
+}
```

Comment:
We probably want to check the contents of `permittedAuthMethodPubkeys` as its likely users won't think to pass a value for pubkey for most auth method types. Or to generalize, make sure that there is equality in collection lengths between the auth method metadata so we don't revert on the transaction if possible.

Reply:
I think it's a good call to add checks for collection lengths to save us some costs on failed TXs.